core/src/main/scala/org/apache/spark/internal/config/History.scala

```diff
@@ -49,6 +49,13 @@ private[spark] object History {
     .timeConf(TimeUnit.SECONDS)
     .createWithDefaultString("7d")
 
+  // This is designed to be smaller than the default value of
+  // `dfs.namenode.fs-limits.max-directory-items` (1024 * 1024).
+  val MAX_LOG_NUM = ConfigBuilder("spark.history.fs.cleaner.maxNum")
```

Comment:
I think this will need to be documented publicly if `spark.history.fs.cleaner.maxAge` is, and document that both limitations apply. Nit: the default isn't quite consistent with the comment above, which says 1024*1024.

Reply:
- Sure for the document.
- The value is designed to be smaller as described at line 52 (`This is designed to be smaller than the default value`) because Spark applications will fail due to event log file creation if the directory contains 1024*1024 files.

Reply:
OK sounds good.